src/api/db/data/20200326221616_regenerate_notifications.rb

```diff
@@ -0,0 +1,98 @@
+#  After some changes in Notification's database structure, some data needs to
+#  be updated. But, instead of fixing the existing Notifications, we are going
+#  to delete the affected ones and regenerate them with the correct values.
+#
+#  Steps:
+#  - Remove notifications without notifiable element. This includes
+#    CommentForPackage and CommentForProject.
+#  - Remove notifications with event_type values ReviewWanted, RequestCreate,
+#    RequestStatechange or CommentForRequest.
+#  - Take all the existing BsRequests in state 'open' and create a RequestCreate
+#    Notification for each of them.
+#  - Take all the existing BsRequests in state 'declined' and create a
+#    RequesStatechange Notification for each of them.
+#  - ?? Take all the existing BsRequests in state 'review' with reviews in state 'new' ??
+#  - Take the BsRequests created in the last 2 weeks and create a CommentForRequest
+#    Notification for each comment on each BsRequest.
+
+class RegenerateNotifications < ActiveRecord::Migration[5.2]
+  def up
+    # delete_obsole_notifications to add here or keep in remove_obsole_notifications?
+    create_request_create_notifications
+    create_request_statechange_notifications
+    create_review_wanted_notifications
+    create_comment_for_request_notifications
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
+  end
+
+  # RequestCreated Notifications
+
+  def create_request_create_notifications
+    new_requests.each do |request|
+      event = Event::RequestCreate.new(request.notify_parameters)
+      create_notifications(event)
+    end
+  end
+
+  def new_requests
```

Comment:
nitpick: Why a method and not a variable? Like

```ruby
new_requests = BsRequest.in_states(:new)
new_requests.each do |request|
...
```

Improves readability IMHO

Reply:
Good advice!